spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainerIntegrationTests.java

```diff
@@ -551,12 +551,16 @@ public void testReplyToConsumersReduced() throws Exception {
 		CachingConnectionFactory cf = new CachingConnectionFactory("localhost");
 		DirectReplyToMessageListenerContainer container = new DirectReplyToMessageListenerContainer(cf);
 		container.setBeanName("reducing");
-		container.setIdleEventInterval(500);
+		container.setIdleEventInterval(100);
 		container.afterPropertiesSet();
 		container.start();
-		ChannelHolder channelHolder = container.getChannelHolder();
-		assertTrue(activeConsumerCount(container, 1));
-		container.releaseConsumerFor(channelHolder, false, null);
+		ChannelHolder channelHolder1 = container.getChannelHolder();
+		ChannelHolder channelHolder2 = container.getChannelHolder();
+		assertTrue(activeConsumerCount(container, 2));
+		container.releaseConsumerFor(channelHolder2, false, null);
+		Thread.sleep(500);
```

Comment:
I understand that this is quick and simple fix, but as experience shows we can't rely on the system clock: no guarantee that CPU will get access to the target process exactly after this timeout. 
In the end we are going to have sporadic test failures.

The best way to sleep with short period and check the state of the component we would like to assert afterward.

If you have some argument to proceed with this simple sleep, let me know here.

Thanks